TimeTrackerUI/MainWindow.xaml.cs

```diff
@@ -24,30 +28,140 @@ public partial class MainWindow : Window
         {
-            SelectionCombobox.Items.Add(AddTextBox.Text);
+            if (!IsTextValid())
+                MessageBox.Show("Please write something to add");
+            else
+            {
+                if (TaskAlreadyExist())
+                    MessageBox.Show("You can not add the same task twice.");
+                else
+                {
+                    SelectionCombobox.Items.Add(AddTextBox.Text.ToUpper());
+                    UpdatePropertiesSettings();
+                }
+                UpdateTasks(AddTextBox.Text.ToUpper());
+                AddTextBox.Clear();
+            }
+        }
+
+        private bool IsTextValid()
+        {
+            if (string.IsNullOrEmpty(AddTextBox.Text))
+                return false;
+            else
+                return true;
+        }
+
+        private bool TaskAlreadyExist()
+        {
+            if (SelectionCombobox.Items.Contains(AddTextBox.Text.ToUpper()))
+                return true;
+            else
+                return false;
+        }
+
+        private void UpdatePropertiesSettings()
+        {
+            Properties.Settings.Default.Combobox.Clear();
+            foreach (var item in SelectionCombobox.Items)
+            {
+                Properties.Settings.Default.Combobox.Add(item.ToString().ToUpper());
+            }
+            Properties.Settings.Default.Save();
         }
 
         private void DeleteButton_Click(object sender, RoutedEventArgs e)
         {
-            SelectionCombobox.Items.RemoveAt(SelectionCombobox.Items.IndexOf(AddTextBox.Text));
+            UpdateTasks(SelectionCombobox.SelectedItem.ToString());
+            SelectionCombobox.Items.Remove(SelectionCombobox.SelectedItem);
+            UpdatePropertiesSettings();
+        }
+
+        private List<Tasks> GetTasks()
+        {
+            for (int i = 0; i < Properties.Settings.Default.Combobox.Count; i++)
+            {
+                var comboboxItem = Properties.Settings.Default.Combobox[i];
+                _tasks.Add(new Tasks(comboboxItem.ToString().ToUpper()));
+            }
+            return _tasks;
+        }
+
+        private void UpdateTasks(string taskName)
+        {
+            var response = _tasks.Find(x => x.Name == taskName);
+            if (response == null)
+                _tasks.Add(new Tasks(taskName));
+            else
+                _tasks.Remove(response);
         }
+
+        private int GetSelection()
+        {
+            return SelectionCombobox.SelectedIndex;
+        }
+
+        private List<Tasks> _tasks = new List<Tasks>();
+        private Tracker _tracker;
+
+        private void Button_Click(object sender, RoutedEventArgs e)
+        {
+            if (SelectionCombobox.SelectedItem == null)
+                MessageBox.Show("Please select a task.");
+            else
+                _tracker.Update(GetSelection());
+        }
+
+        private void CheckLogButton_Click(object sender, RoutedEventArgs e)
```

Comment:
Do you need the logic of this method here? Can you maybe put it somewhere else? For example inside the tracker ?
Think about it